### ISSUE TITLE
Vary:Accept-Encoding repeated on ipro & pagespeed resources once optimized.

### ISSUE THREAD

**Issue by jmarantz** [pagespeed/ngx_pagespeed#1271]
```
% wget -q --save-headers -O - http://ngxpagespeed.com/styles/yellow.css+blue.css+big.css+bold.css.pagespeed.cc.xo4He3_gYf.css|grep Vary
Vary: Accept-Encoding
Vary: Accept-Encoding
% wget -q --save-headers -O - http://ngxpagespeed.com/styles/yellow.css|grep Vary                                
Vary: Accept-Encoding
% wget -q --save-headers -O - http://ngxpagespeed.com/styles/yellow.css | grep Vary
Vary: Accept-Encoding
Vary: Accept-Encoding
%
```

Note that the first request for yellow.css had only one Vary:AcceptEncoding; I think because it wasn't fresh in the cache.

**Comment by devgraal** [comment]
Using nginx-1.11.5 & pagespeed-1.11.33.4 worked perfectly on debian jessie. The only thing I had to deal with when building from source was to update the nginx-lua module to its latest release, and using openssl-1.0.2j sources to get http2 working.

But, because there is a "but": I am getting two (or more) "Vary" headers on at least CSS, JS & pictures.

I tried disabling gzip_vary without any more success :(

Is there a way to fix that ?

**Comment by jmarantz** [comment]
I'm not saying this isn't a bug, but I'm curious what negative effect you see from duplicate Vary headers.

**Comment by devgraal** [comment]
Even if it is really nothing important, GTmetrix seems to not understand the presence of those 2 headers (being totally identical). So the check has a negative effect on the whole score.
The second "negative" point is that as those 2 headers are sent from the server meaning there is at least 21 chars written "for nothing", and for each object being sent to the browser.

The goal of PageSpeed being to optimize network transfers, I am thinking those ressources could also be optimized ;)

**Comment by jmarantz** [comment]
agreed on both points.  I *think* this might be fixed on trunk already, and we are hoping for a release that includes those fixes real soon now.  I was just trying to measure the urgency.
